File: src/main/java/ux/panel/BasicPanel.java
package ux.panel;

import ux.informer.DrawingInfoSupplier;

public class BasicPanel extends Panel {
    protected BasicPanel(Builder builder) {
        super(builder);
    }

    @Override
    protected void panelSubclassDraw() {
        ;
    }

    public static class Builder extends Panel.Builder<Builder> {

        public Builder(DrawingInfoSupplier drawingInfoSupplier, AlignHorizontal alignHorizontal, AlignVertical vAlign, int width, int height) {
            super(drawingInfoSupplier, alignHorizontal, vAlign, width, height);
        }

        @Override
        public BasicPanel build() {
            return new BasicPanel(this);
        }

        @Override
        protected Builder self() {
            return this;
        }
    }
}

File: src/main/kotlin/ux/panel/BasicPanel.kt
package ux.panel

import ux.informer.DrawingInfoSupplier

class BasicPanel private constructor(builder: Builder?) : Panel(builder!!) {
    override fun panelSubclassDraw() {
    }

    class Builder(
        drawingInfoSupplier: DrawingInfoSupplier?,
        alignHorizontal: AlignHorizontal?,
        vAlign: AlignVertical?,
        width: Int,
        height: Int
    ) : Panel.Builder<Builder?>(
        drawingInfoSupplier!!, alignHorizontal!!, vAlign!!, width, height
    ) {
        override fun build(): BasicPanel {
            return BasicPanel(this)
        }

        override fun self(): Builder {
            return this
        }
    }
}